pep-0517.txt

```diff
@@ -164,74 +164,15 @@ with tools that do not use this spec.
 
-  def get_build_wheel_requires(config_settings):
-      ...
-
-This hook MUST return an additional list of strings containing PEP 508
-dependency specifications, above and beyond those specified in the
-``pyproject.toml`` file, to be installed when building a wheel. Example::
-
-  def get_build_wheel_requires(config_settings):
-      return ["wheel >= 0.25", "setuptools"]
-
-Optional. If not defined, the default implementation is equivalent to
-``return []``.
+Mandatory hooks
+===============
 
 ::
 
-  def prepare_wheel_metadata(metadata_directory, config_settings):
-      ...
-
-Must create a ``.dist-info`` directory containing wheel metadata
-inside the specified ``metadata_directory`` (i.e., creates a directory
-like ``{metadata_directory}/{package}-{version}.dist-info/``. This
-directory MUST be a valid ``.dist-info`` directory as defined in the
-wheel specification, except that it need not contain ``RECORD`` or
-signatures. The hook MAY also create other files inside this
-directory, and a build frontend MUST ignore such files; the intention
-here is that in cases where the metadata depends on build-time
-decisions, the build backend may need to record these decisions in
-some convenient format for re-use by the actual wheel-building step.
-
-This must return the basename (not the full path) of the ``.dist-info``
-directory it creates, as a unicode string.
-
-Optional. If a build frontend needs this information and the method is
-not defined, it should call ``build_wheel`` and look at the resulting
-metadata directly.
-
-::
-
-  def prepare_wheel_build_files(build_directory, config_settings):
-      ...
-
-Must copy or create any files needed to build a wheel of this package into
-``build_directory``. For instance, ``pyproject.toml`` should
-be copied unmodified into the root of this directory. For tools such
-as `setuptools_scm <https://github.com/pypa/setuptools_scm>`_, this may include
-extracting some information from a version control system.
-The ``build_wheel`` hook will subsequently be run from the ``build_directory``
-populated by this hook. The contents of the resulting wheel should be the same
-whether ``build_wheel`` is invoked in an original source directory, the build
-directory populated by this hook, or an unpacked sdist directory.
-
-Because the wheel will be built from a temporary build directory, ``build_wheel``
-may create intermediate files in the working directory, and does not need to
-take care to clean them up.
-
-The return value will be ignored.
-
-Optional. If this hook is not defined, frontends may call ``build_sdist``
-and unpack the archive to use as a build directory. Backends in which
-building an sdist has additional requirements should define
-``prepare_wheel_build_files``.
-
-::
-
-  def build_wheel(wheel_directory, config_settings, metadata_directory=None):
-      ...
+    build_wheel(wheel_directory, config_settings={}, build_directory=None, metadata_directory=None):
```

Comment:
`config_settings=None` would be a more conventional spelling for "use the default build settings" than using a mutable default value.